src/components/form/date-input/date-input.component.ts

```diff
@@ -18,38 +18,66 @@ import { ControlValueAccessor, NG_VALIDATORS, NG_VALUE_ACCESSOR, Validator } fro
     }
   ],
 })
-export class DateInputComponent implements ControlValueAccessor, Validator, OnDestroy {
+export class DateInputComponent implements ControlValueAccessor, Validator, OnInit, OnDestroy {
   @Input()
   public id: string;
 
   @Input()
   public mandatory: boolean;
 
+  @Input()
+  public dateTime: boolean;
+
   @Input()
   public formControl: FormControl;
 
   public displayDay: string = null;
   public displayMonth: string = null;
   public displayYear: string = null;
 
+  public displayHour: string = null;
+  public displayMinute: string = null;
+  public displaySecond: string = null;
   // tslint:disable-next-line
-  private readonly DATE_FORMAT = /^(\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$/;
-
+  // private readonly DATE_FORMAT = /^(\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$/;
+  private readonly DATE_FORMAT =
+    /^(\d{4})-?(\d\d)-?(\d\d)(?:T(\d?\d)(?::?(\d?\d)(?::?(\d\d)(?:\.(\d+))?)?)?(Z|([+-])(\d\d):?(\d\d))?|Z)?$/;
+//    1        2       3         4          5          6          7          8  9     10      11
   private propagateChange: (_: any) => {};
   private rawValue: string = '';
   private day: string;
   private month: string;
   private year: string;
+  private hour: string;
+  private minute: string;
+  private second: string;
+
+  public ngOnInit() {
+    if (this.mandatory) {
+      this.displayHour = '00';
+      this.displayMinute = '00';
+      this.displaySecond = '00';
+      this.hour = '00';
+      this.minute = '00';
+      this.second = '00';
+    }
+  }
 
-  public writeValue(obj: string): void {
+  public writeValue(obj: string): void { // 2018-04-09T08:02:27.542
     if (obj) {
-      this.rawValue = obj.replace(/T.*/, '');
-
+      this.rawValue = obj.replace(/\..*/, '');
       // needs to handle also partial dates, e.g. -05-2016 (missing day)
-      const values = this.rawValue.split('-');
-      this.year = this.displayYear = values[0] || '';
-      this.month = this.displayMonth = values[1] || '';
-      this.day = this.displayDay = values[2] || '';
+      const valueParts = this.rawValue.split('T');
+      const dateValues = valueParts[0].split('-');
```

Comment:
can we use es6 desctructuring: 

[date, time] = this.rawValue.split('T');
dateValues = date.split('-');
...
if(time)
timeParts = time.split(':');
...